components/PlantDetails.js

```diff
@@ -0,0 +1,84 @@
+import Image from "next/image";
+import Link from "next/link";
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import useLocalStorageState from "use-local-storage-state";
+
+const lightNeedIcon = {
+  "Full Sun": "☀️☀️☀️☀️",
+  "Partial Shade": "☀️☀️☀️",
+  Shade: "☀️☀️",
+  "Full Shade": "☀️",
+};
+
+const waterNeedIcon = {
+  Low: "💧",
+  Medium: "💧💧",
+  High: "💧💧💧",
+};
+
+const seasonIcons = {
+  Spring: "🌱",
+  Summer: "🐝",
+  Fall: "🍂",
+  Winter: "❄️",
+};
+
+export default function PlantDetails() {
+  const router = useRouter();
+  const { id } = router.query;
```

Comment:
Keep the component as dumb as possible. Do all the router logic better in the page 'pages/[id].js'. There you can grab the id from the router, search for the specific plant and pass the right one as a prop to the PlantDetails component.